python/pyarrow/src/arrow/python/deserialize.cc

```diff
@@ -321,183 +321,5 @@ Status DeserializeSet(PyObject* context, const Array& array, int64_t start_idx,
       out);
 }
 
```

Comment:
Is the rest of the file still being used? I would expect us to entirely remove `deserialize.h` and `deserialize.cc`.

Reply:
`SparseTensorCounts` is exposed and defined in [libarrow_python.pxd](https://github.com/apache/arrow/blob/b3d218c819283bafe973dc7deb5214324f4a68b2/python/pyarrow/includes/libarrow_python.pxd#L219) though not used anywhere except in the removed `GetSerializedFromComponents`. Should it also be deprecated first?

Reply:
It is similar for others so I will remove files altogether.